fastpair/rust/demo/rust/src/fetcher/common.rs

```diff
@@ -41,36 +25,26 @@ pub(crate) trait FpFetcher {
     ) -> Result<DeviceInfo, anyhow::Error>;
 }
 
-/// A unit struct for retrieving Fast Pair information from the local filesystem.
-pub(crate) struct FpFetcherLocal {
-    path: String,
+/// Holds Fast Pair device information parsed from JSON.
+#[derive(Clone, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub(crate) struct DeviceInfo {
+    image_url: String,
+    name: String,
 }
 
-impl FpFetcherLocal {
-    pub(crate) fn new(path: String) -> Self {
-        FpFetcherLocal { path }
-    }
+/// Holds top-level Fast Pair information parsed from JSON. See `local`
+/// directory for format.
+#[derive(Deserialize)]
+pub(super) struct JsonData {
```

Comment:
if it's just one member, it's probably worth just making this a tuple struct, e.g.

pub(super) struct JsonData(DeviceInfo). not sure how serde will handle that though.